### DISCUSSION TITLE
【求助】打包云原神后，鼠标不能操作场景（

### DISCUSSION THREAD

**Discussion by Musicminion** [tw93/Pake#587] [Q&A] [answered]
抱歉打扰到开发者们了！ 自己想打包一个macOS的云原神（~~为了让启动更有仪式感~~）
- 云原神的地址：https://ys.mihoyo.com/cloud
- 遇到的问题：直接在浏览器玩的时候，点击游戏内部，是可以把鼠标给**吞掉**的，就是鼠标无论移动，可以操作游戏里面视野的移动，但是鼠标**不会跑出来**，只有按了特定的快捷键后，鼠标才可以跑出来。用了Pake打包之后，鼠标就可以随意的移动进去、移动出来，移动的时候视野就没有任何转动，想问一下这种可能是什么原因，或者有什么方法呢？

不知道有没有大佬能给一点提示呢？

**Comment by Tlntin** [comment]
可能是鼠标没有被捕捉

**Comment by tw93** [comment]
这里可以使用系统默认浏览器，比如说mac下面的Safari试试，应该会有一样的效果复现

**Reply by Musicminion** [reply]
我用Safari玩的时候没有鼠标捕捉不了的问题诶，有点神奇，我又测了一下用PWA安装，然后发现没什么问题，可惜用Pake打包就是不能捕捉鼠标，有点奇怪，【理论上来说是在程序里面点一下，然后鼠标就消失了，然后就可以通过鼠标来操作场景视野，如果按esc，鼠标就会出现】想了解一下捕捉鼠标的js代码一般用什么呢（

**Comment by jeasonnow** [comment]
看起来是个值得研究的话题，感觉问题在于 Pake 打包出来的 app 没有实现“点击**游戏内部**，可以把鼠标给吞掉（设置页面整体 cursor），并在内部实现鼠标移动路径到游戏的映射“

**Reply by jeasonnow** [reply]
我晚点用开发模式打一个看看是不是什么 API 不支持

**Reply by jeasonnow** [reply]
https://developer.mozilla.org/zh-CN/docs/Web/API/Pointer_Lock_API 目前看起来很可能跟这个 API 是否被正常调用强相关，云游戏本身依托于 webrtc 和对外部 IO 设备操作的转发，现在看来云原神是基于这个 API 来进行鼠标锁定（隐藏）和后续鼠标滚动转发的。

**Reply by jeasonnow** [reply]
![image](https://github.com/tw93/Pake/assets/16950207/c868dec7-f853-47a6-8ee4-41af16353fbb)
![image](https://github.com/tw93/Pake/assets/16950207/afe67cf2-a8b6-46b8-9b95-e92a24720e60)
我在官方找到了同样的问题，看起来可能和内核和 Tauri 的实现有关：
https://github.com/tauri-apps/wry/issues/445
https://github.com/tauri-apps/tauri/issues/5118

**Reply by jeasonnow** [reply]
https://tauri.app/zh-cn/v1/api/js/window/#setcursorgrab

感觉我们可以做个兼容。

**Reply by Musicminion** [reply]
辛苦大佬找bug了！前端小菜鸡只能眼巴巴羡慕大佬们惹（

**Comment by Musicminion** [comment, marked as answer]
今天意外用electron打包，居然可以了，视野追踪、快捷键什么的都可以，不得不说真的很神奇（
